### PR TITLE
canvas: Cleanup CanvasData and layout and script messages.

### PR DESCRIPTION
**Pull request by emilio**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] These changes do not require tests because refactoring.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14099)

### CODE REVIEW

**Comment by emilio** [comment]
This fixes #13734.

File: components/script/dom/htmlcanvaselement.rs [hunk 1]
```diff
@@ -202,16 +202,17 @@ impl HTMLCanvasElement {
...
             match receiver.recv().unwrap() {
-                CanvasData::Pixels(pixel_data)
-                    => pixel_data.image_data.to_vec(),
-                CanvasData::WebGL(_)
-                    // TODO(emilio): Not sure if WebGL canvas is required for 2d spec,
-                    // but I think it's not.
-                    => return None,
+                Some(pixels) => pixels,
+                None => {
+                    // TODO(emilio): Not sure if WebGL canvas is required for 2d
```
Comment (Ms2ger) [inline components/script/dom/htmlcanvaselement.rs]: I see no reason why it would not be required; please file an issue.

Comment (emilio) [inline components/script/dom/htmlcanvaselement.rs]: Hmm, IIRC the last time I looked at this the canvas 2d spec only defined the behavior when another 2d canvas was provided, but yeah, worth investigating and fixing if required.

**Comment by emilio** [comment]
@bors-servo r=nox

 * Updated the comment to point to #14109

**Pull request merged**